src/intobj.h

```diff
@@ -111,6 +111,25 @@ static inline Int INT_INTOBJ(Obj o)
 #endif
 }
 
+/****************************************************************************
+**
+*F  INT_CAN_BE_INTOBJ( <i> ) . check if a C integer fits in an immediate
+**  integer.
+*/
+static inline Int INT_CAN_BE_INTOBJ(Int i)
+{
+    return (-(1L<<NR_SMALL_INT_BITS) <= i) && (i < 1L<<NR_SMALL_INT_BITS );
+}
+
+/****************************************************************************
+**
+*F  UINT_CAN_BE_INTOBJ( <i> ) . check if a C unsigned integer fits in an
+**  immediate integer.
+*/
+static inline Int UINT_CAN_BE_INTOBJ(UInt i)
+{
+    return i < 1UL<<NR_SMALL_INT_BITS;
```

Comment:
Off-topic remark: Strictly speaking this is not correct, as 1UL indicates an `unsigned long`, but there are 64bit systems were long is still only 32bits (you need `long long` for a 64bit type). Specifically, the LLP64 data model, which is what Windows uses on 64bit systems.

The same of course also holds for the 1L above. But of course we already use 1L and 1UL blissfully in lots of places, relying implicitly on all futures 64-systems GAP will run on to use LP64 (which is what Linux, *BSD, OSX use). Portable would be to use the `INT64_C` and `U INT64_C` etc. macros, resp. provide our own `UINT_C` macro (which maps to either `UINT64_C` or `UINT32_C` depending on the host bit size.

But again, we already do this wrong in a ton of places, so this PR is not the place to start working on it, I just wanted to mention it because reading that line reminded me of it.